### PR TITLE
Move RemoteClusterService into TransportService

### PR DESCRIPTION
**Pull request by s1monw**
TransportService and RemoteClusterService are closely coupled already today
and to simplify remote cluster integration down the road it can be a direct
dependency of TransportService. This change moves RemoteClusterService into
TransportService with the goal to make it a hidden implementation detail
of TransportService in followup changes.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/transport/TransportService.java [hunk 1]
```diff
@@ -209,6 +222,10 @@ protected void doStart() {
             false, false,
             (request, channel) -> channel.sendResponse(
                     new HandshakeResponse(localNode, clusterName, localNode.getVersion())));
+        if (remoteClusterService != null) {
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: out of curiosity, when can it be null?

File: core/src/main/java/org/elasticsearch/transport/TransportService.java [hunk 2]
```diff
@@ -158,12 +161,22 @@ public TransportService(Settings settings, Transport transport, ThreadPool threa
...
         this.interceptor = transportInterceptor;
         this.asyncSender = interceptor.interceptSender(this::sendRequestInternal);
+        final boolean connectToRemoteCluster = RemoteClusterService.ENABLE_REMOTE_CLUSTERS.get(settings);
+        remoteClusterService = new RemoteClusterService(settings, this);
         if (clusterSettings != null) {
             clusterSettings.addSettingsUpdateConsumer(TRACE_LOG_INCLUDE_SETTING, this::setTracerLogInclude);
             clusterSettings.addSettingsUpdateConsumer(TRACE_LOG_EXCLUDE_SETTING, this::setTracerLogExclude);
+            if (connectToRemoteCluster) {
+                clusterSettings.addAffixUpdateConsumer(RemoteClusterService.REMOTE_CLUSTERS_SEEDS,
+                    remoteClusterService::updateRemoteCluster, (namespace, value) -> {});
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: should we be calling listenForUpdates here ? where is it called now?

**Review by javanna — APPROVED** [review]
left a comment or two, LGTM otherwise

File: core/src/main/java/org/elasticsearch/transport/TransportService.java (hunk 2 above)
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: oh I didn't update after I merged...

File: core/src/main/java/org/elasticsearch/transport/TransportService.java (hunk 1 above)
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/transport/TransportService.java]: it used to be in an earlier commit... I will remove

**Pull request merged**